pkg/steps/testing.go

```diff
@@ -37,10 +71,28 @@ func (dl *DryLogger) AddObject(o runtime.Object) {
 	dl.objects = append(dl.objects, o)
 }
 
-// GetObjects returns the list of objects.
+// GetObjects returns the list of decoded objects with kind and version included.
 func (dl *DryLogger) GetObjects() []runtime.Object {
 	dl.RLock()
 	defer dl.RUnlock()
+
+	for _, object := range dl.objects {
```

Comment:
Why is this necessary?

Reply:
We need to encode/decode each object to include the missing `Kind` `APIVersion` etc fields.

Reply:
What uses `GetObjects`? I would just expect the logger to encode once on output

Reply:
I did the encoding/decoding here to iterate in the objects just once.

Reply:
therefore, in the `main()` we can just call the `.GetObjects()` and don't care about the encode/decode and the order of the objects.

Reply:
I agree with :steve: here. As I said elsewhere about `GetObjects`, the API is a bit strange for a type called `DryLogger`.

I would expect something to the effect of

```go
func (dl *DryLogger) Log(w io.Writer) error {
 	dl.RLock()
    objs := dl.GetObjects()
    dl.RUnlock() 
    // encode/decode/order
    // log
}
```

Reply:
addressed